sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala

```diff
@@ -2740,6 +2740,19 @@ object SQLConf {
       .booleanConf
       .createWithDefault(false)
 
+  val UNION_BYNAME_STRUCT_SUPPORT_ENABLED =
```

Comment:
I'm not sure this worths a config. It's natural that the `byName` and `allowMissingCol` flag should apply to nested column, and these 2 flags are newly added in the master branch so there is no backward compatibility issues.

Reply:
Ok, sounds more correct. I will remove this config.